src/Microsoft.AspNet.Mvc.Core/SessionStateTempDataProvider.cs

```diff
@@ -61,11 +61,16 @@ public class SessionStateTempDataProvider : ITempDataProvider
                 return null;
             }
 
-            var tempDataDictionary = new Dictionary<string, object>(StringComparer.OrdinalIgnoreCase);
             var session = context.Session;
+            if (session == null)
```

Comment:
Discussed offline: we should not throw if Session is not available as it would impact users who do not even care about TempData(example: purely Web API users). Even logging the details might not help as we potentially might need to log it as a Warning and it would also appear in logs when the user doesn't care about TempData to begin with.